src/main/java/uk/gov/hmcts/reform/managecase/service/CaseAssignmentService.java

```diff
@@ -138,13 +144,25 @@ private CaseDetails getCase(CaseAssignment input) {
             .collect(toList());
     }
 
-    private Optional<ProfessionalUser> findUserBy(List<ProfessionalUser> users, String assigneeId) {
-        return users.stream()
-                .filter(user -> assigneeId.equalsIgnoreCase(user.getUserIdentifier()))
-                .findFirst();
+    private boolean isAssigneePresent(List<ProfessionalUser> users, String assigneeId) {
+        return users.stream().anyMatch(user -> assigneeId.equalsIgnoreCase(user.getUserIdentifier()));
     }
 
     private boolean containsIgnoreCase(List<String> list, String value) {
         return list.stream().anyMatch(value::equalsIgnoreCase);
     }
+
+    private List<String> getAssigneeRoles(String assigneeId) {
+        String systemUserToken = idamRepository.getSystemUserAccessToken();
+        List<uk.gov.hmcts.reform.idam.client.models.UserDetails> users =
+                idamRepository.searchUsers(systemUserToken, String.format(IDAM_ES_QUERY, assigneeId));
+
+        return users.stream()
+                .filter(user -> assigneeId.equalsIgnoreCase(user.getId()))
+                .reduce((a, b) -> {
+                    throw new IllegalStateException("Multiple users with same IDAM with id:: " + assigneeId);
+                })
+                .get()
+                .getRoles();
```

Comment:
with the above-described improvement in repository implementation, this line could read much simpler.